docs/collections.md

```diff
@@ -0,0 +1,333 @@
+# Sprinkler: Collections
+
+## Intentions
+
+The purpose of this module is to provide read-only collections to the user.
+
+Then why not use [kotlinx.collections.immutable](https://github.com/Kotlin/kotlinx.collections.immutable) instead?
+
+Because while this library offers a good API, it is not cast-safe in certain places. Meaning that even though one
+created a supposed immutable collection, one can still mutate it by casting it to a mutable collection beforehand
+(this is particularly notable with iterators).
+
+Moreover, for semantic reasons, the collections here are not called `Immutable*` but `ReadOnly*` instead.
+
+## API Documentation
+
+The API is a general mirror of the standard library's collections API, 
+but prepended with `ReadOnly` or `readOnly` on its symbols.
+
+### Summary
```

Comment:
Too much details, maybe just a [table](https://github.com/Kotlin/kotlinx.collections.immutable/blob/master/README.md#interfaces-and-implementations) comparing each interface with its basic version would be enough. If you really want it, you can present 1 implementation (e.g. collection) in details, but more is redundant.

You can also add a short section about the extensions and the difference between the **toReadOnlyX** **and the asReadOnlyX** methods and that would be good imo.

Reply:
I left the complete api documentation, but i added a more general, summarized, explanation